controller/deployments.go

```diff
@@ -410,7 +314,8 @@ func (c *DeploymentsController) ShowDeploymentStats(ctx *app.ShowDeploymentStats
 
 	deploymentStats, err := kc.GetDeploymentStats(*kubeSpaceName, ctx.AppName, ctx.DeployName, startTime)
 	if err != nil {
-		return errors.NewInternalError(ctx, errs.Wrapf(err, "could not retrieve deployment statistics for %s", ctx.DeployName))
+		return errors.NewInternalError(ctx,
```

Comment:
what about returning a `404 Not Found` instead of a `500` error, or just `wrapf` the error ?

Reply:
I think the 404 is treated in the lines after this change, and I'd like to distinguish it from any other errors. 

I used NewInternalError to make it explicit what the input arguments were that led to the error.
I'm happy to simply return the error instead.

Reply:
my point is, the `GetDeploymentStats` function should return a `NotFoundError` and you should not need to convert it in an `InternalError`.